internal/services/cdn/cdn_frontdoor_origin_resource.go

```diff
@@ -320,24 +333,29 @@ func resourceCdnFrontDoorOriginUpdate(d *pluginsdk.ResourceData, meta interface{
 	if d.HasChange("private_link") {
 		// I need to get the profile SKU so I know if it is valid or not to define a private link as
 		// private links are only allowed in the premium sku...
-		profileId := parse.NewFrontDoorProfileID(id.SubscriptionId, id.ResourceGroup, id.ProfileName)
-		profile, err := profileClient.Get(ctx, profileId.ResourceGroup, profileId.ProfileName)
+		profileId := profiles.NewProfileID(id.SubscriptionId, id.ResourceGroup, id.ProfileName)
+
+		profileResp, err := profileClient.Get(ctx, profileId)
 		if err != nil {
-			if utils.ResponseWasNotFound(profile.Response) {
-				return fmt.Errorf("retrieving parent %s: not found", profileId)
+			if response.WasNotFound(profileResp.HttpResponse) {
+				d.SetId("")
+				return nil
```

Comment:
This is unusual behaviour for an `Update`, if it's not found, `Read` on this resource (probably the parent actually) should have caught it and removed it from state? Is this a copy-paste error perhaps?
```suggestion
			if response.WasNotFound(profileResp.HttpResponse) {
				return fmt.Errorf("retrieving parent %s: not found", profileId)
```

Reply:
May have been... fixed.